packages/core/src/Actor.ts

```diff
@@ -1,59 +1,126 @@
+): ActorRef<TEvent> {
+  return new ObservableActorRef(
+    createMachineBehavior(machine, parent, options),
+    name
+  );
+}
+
+export function fromService<TContext, TEvent extends EventObject>(
+  service: Interpreter<TContext, any, TEvent>,
+  name: string = registry.bookId()
+): ActorRef<TEvent> {
+  return new ObservableActorRef(createServiceBehavior(service), name);
+}
+
+enum ProcessingStatus {
+  NotProcessing,
+  Processing
+}
+
+export class ObservableActorRef<TEvent extends EventObject, TEmitted>
+  implements ActorRef<TEvent, TEmitted> {
+  public current: TEmitted;
+  private context: ActorContext;
+  private behavior: Behavior<TEvent, TEmitted>;
+  private mailbox: TEvent[] = [];
+  private processingStatus: ProcessingStatus = ProcessingStatus.NotProcessing;
+  public name: string;
+
+  constructor(behavior: Behavior<TEvent, TEmitted>, name: string) {
+    this.behavior = behavior;
+    this.name = name;
+    this.context = {
+      self: this,
+      name: this.name
+    };
+    this.current = behavior.current;
+  }
+  public start() {
+    this.behavior = this.behavior.receiveSignal(this.context, startSignal);
+    return this;
+  }
+  public stop() {
+    this.behavior = this.behavior.receiveSignal(this.context, stopSignal);
+  }
+  public subscribe(observer) {
+    return this.behavior.subscribe?.(observer) || nullSubscription;
+  }
+  public send(event) {
+    this.mailbox.push(event);
+    if (this.processingStatus === ProcessingStatus.NotProcessing) {
+      this.flush();
+    }
+  }
+  private flush() {
+    while (this.mailbox.length) {
+      const event = this.mailbox.shift()!;
 
-export interface Actor<
-  TContext = any,
-  TEvent extends EventObject = AnyEventObject
-> extends Subscribable<TContext> {
-  id: string;
-  send: (event: TEvent) => any; // TODO: change to void
-  stop?: () => any | undefined;
-  toJSON: () => {
-    id: string;
-  };
-  meta?: InvokeDefinition<TContext, TEvent>;
-  state?: any;
-}
-
-export function createNullActor<TContext, TEvent extends EventObject>(
-  id: string
-): Actor<TContext, TEvent> {
-  return {
-    id,
-    send: () => void 0,
-    subscribe: () => ({
-      unsubscribe: () => void 0
-    }),
-    toJSON: () => ({
-      id
-    })
-  };
-}
-
-/**
- * Creates a null actor that is able to be invoked given the provided
- * invocation information in its `.meta` value.
- *
- * @param invokeDefinition The meta information needed to invoke the actor.
- */
-export function createInvocableActor<TContext, TEvent extends EventObject>(
-  invokeDefinition: InvokeDefinition<TContext, TEvent>
-): Actor<any, TEvent> {
-  const tempActor = createNullActor<TContext, TEvent>(invokeDefinition.id);
-
-  tempActor.meta = invokeDefinition;
-
-  return tempActor;
-}
-
-export function isActor(item: any): item is Actor {
-  try {
-    return typeof item.send === 'function';
-  } catch (e) {
-    return false;
+      this.processingStatus = ProcessingStatus.Processing;
```

Comment:
nit: setting `processingStatus` could be moved outside of this loop, it would imho illustrate the intent slightly better